src/engine/pagination.ts

```diff
@@ -0,0 +1,107 @@
+import { queryValueToIntOrUndefined } from '../utils'
+import {
+  Query,
+  AbstractQuery,
+  ParseQuery,
+  useAbstractQuery,
+  Router,
+  AbstractQueryOptions,
+} from './query'
+
+export type PaginationQuery = {
+  /**
+   * The page index.
```

Comment:
Could you add the default value here and for the size as well?

Reply:
Done